src/control/server/mgmt_system.go

```diff
@@ -1534,6 +1544,11 @@ func (svc *mgmtSvc) selfHealNotifyPSes(ctx context.Context, propVal string) erro
 			ID:         id,
 			SysPropVal: propVal,
 		}
+		// Set request hostlist from system request so control-API pool call reaches
+		// MS-replicas and includes any custom server port assignments (as we don't have
+		// access to the server config from here).
```

Comment:
This is a bit odd to me. Shouldn't we have the MS addresses with port saved in the mgmt db, at the very least?